### ISSUE TITLE
`User-Agent`, `Accept-Encoding` and `Host` headers

### ISSUE THREAD

**Issue by gulien** [gotenberg/gotenberg#900]
I tried to generate a PDF using a Gotenberg container, but the resulting file was completely blank. I sent a POST request with [this URL](https://www.sec.gov/Archives/edgar/data/1067983/000119312524133049/d759664d8k.htm), but all I got was a white, empty page. You need to set the HTTP headers like [this](https://www.sec.gov/os/accessing-edgar-data). However, when I manually downloaded the HTML file and sent it to my Gotenberg container, the conversion worked like a charm. This made me think that maybe the `extraHttpHeaders` parameter isn't working as it should for `User-Agent`, `Accept-Encoding` and `Host`.

To get to the bottom of this, I was wondering if there's a way to take a closer look at the actual headers and body that a Gotenberg instance sends when it makes a GET request.

_Originally posted by @wjkoh in https://github.com/gotenberg/gotenberg/discussions/842#discussioncomment-9752869_

**Comment by gulien** [comment]
@wjkoh

Looks like:

* `User-Agent`: available via https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setUserAgentOverride
* `Accept-Encoding`: experimental -> https://chromedevtools.github.io/devtools-protocol/tot/Network/#method-setAcceptedEncodings
* `Host`: not available -> https://github.com/puppeteer/puppeteer/issues/6167#issuecomment-1327349162

**Comment by gulien** [comment]
I wonder if, alternatively, you could download the file using your language client and then send it to Gotenberg as PDF. No ideal though. I could make the first two headers work, but not the `Host` one.

**Comment by wjkoh** [comment]
> I wonder if, alternatively, you could download the file using your language client and then send it to Gotenberg as PDF. No ideal though. I could make the first two headers work, but not the `Host` one.
> 
> EDIT: it seems that only the `User-Agent` is required -> https://www.sec.gov/os/accessing-edgar-data

Yes, that's correct. After conducting a few tests, I discovered that the server is only particular about the User-Agent header. This is great news! If you can implement the User-Agent override, it would be incredibly helpful for me and most users working on HTML-to-PDF conversion tasks.
